test/src/bridge/lib_core_test.dart

```diff
@@ -9,201 +9,338 @@ import 'package:flutter_ble_lib/src/_internal.dart';
-    Map<dynamic, dynamic> decodedCharacteristic = decodedRoot["characteristic"];
+    Map<String, dynamic> decodedRoot
+  ) {
+    Map<String, dynamic> decodedCharacteristic = decodedRoot["characteristic"];
     String transactionId = decodedRoot["transactionId"];
     return CharacteristicWithValueAndTransactionId.fromJson(
       decodedCharacteristic,
-      Service.fromJson(decodedRoot, peripheral, null),
-      null,
+      Service.fromJson(decodedRoot, peripheral, MockInternalBleManager()),
+      MockInternalBleManager(),
     ).setTransactionId(transactionId);
   }
 
-  Map<dynamic, dynamic> createRawCharacteristic(
-          {int id,
-          int serviceId,
-          String serviceUuid,
-          String characteristicUuid,
-          String transactionId,
-          String base64value}) =>
-      <String, dynamic>{
-        "serviceUuid": serviceUuid,
-        "serviceId": serviceId,
-        "transactionId": transactionId,
-        "characteristic": <String, dynamic>{
-          "characteristicUuid": characteristicUuid,
-          "id": id,
-          "isReadable": true,
-          "isWritableWithResponse": false,
-          "isWritableWithoutResponse": false,
-          "isNotifiable": true,
-          "isIndicatable": false,
-          "value": base64value ?? ""
-        }
-      };
+  Map<String, dynamic> createRawCharacteristic(
+    {
+    bool mockIds = false,
+    int? id,
+    int? serviceId,
+    String? serviceUuid,
+    String? characteristicUuid,
+    String? transactionId,
+    String base64value = ""
+  }) { 
+    if (mockIds) {
+      serviceUuid ??= mockServiceUuid;
+      serviceId ??= mockServiceId;
+      characteristicUuid ??= mockCharacteristicUuid;
+      id ??= mockCharacteristicId;
+    }
+
+    return <String, dynamic>{
+          "serviceUuid": serviceUuid,
+          "serviceId": serviceId,
+          "transactionId": transactionId,
+          "characteristic": <String, dynamic>{
+            "characteristicUuid": characteristicUuid,
+            "id": id,
+            "isReadable": true,
+            "isWritableWithResponse": false,
+            "isWritableWithoutResponse": false,
+            "isNotifiable": true,
+            "isIndicatable": false,
+            "value": base64value,
+          }
+        };
+  }
 
   test('monitorCharacteristicForIdentifier cancels on stream error', () async {
-    expectLater(
-        bleLib.monitorCharacteristicForIdentifier(peripheral, 123, null),
-        emitsInOrder([
-          emitsError(isInstanceOf<BleError>()),
-          emitsDone,
-        ]));
-    await emitPlatformError(cancellationErrorJson);
+    final mockTransId = "asdasd";
+
+    final fut = expectLater(
+      bleLib.monitorCharacteristicForIdentifier(peripheral, 123, mockTransId),
+      emitsInOrder([
+        emitsError(isInstanceOf<BleError>()),
+        emitsDone,
+      ])
+    );
+    await _emitPlatformError(cancellationErrorJson(mockTransId));
+
+    // await fut;
+    print("anyad");
```

Comment:
Remove, please

Reply:
:D ups... it was hard fixing tests, so... :D